adapters_test.go

```diff
@@ -0,0 +1,240 @@
+	require.NoError(t, err)
+	cosmosEthTxBytes, err := cosmosEthTx.MarshalBinary()
+	require.NoError(t, err)
+	invalidData := hexutil.Bytes("invalid")
+
+	t.Run("returns error when input tx contains invalid binary data", func(t *testing.T) {
+		_, err := monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{invalidData}, nil, "")
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when no deposit txs are present", func(t *testing.T) {
+		_, err = monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{cosmosEthTxBytes}, nil, "")
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when signing tx fails", func(t *testing.T) {
+		_, err = monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{hexutil.Bytes(depositTxBytes)}, func(_ *tx.Tx) error {
+			return errors.New("sign tx error")
+		}, "")
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unable to unmarshal binary tx data", func(t *testing.T) {
+		_, err = monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{hexutil.Bytes(depositTxBytes), hexutil.Bytes(cosmosEthTxBytes), invalidData}, nil, "")
+		require.Error(t, err)
+	})
+}
+
+func TestAdaptCosmosTxsToEthTxs(t *testing.T) {
+	t.Run("returns empty slice when input txs is nil", func(t *testing.T) {
+		txs, err := monomer.AdaptCosmosTxsToEthTxs(nil)
+		require.NoError(t, err)
+		require.Empty(t, txs)
+	})
+
+	t.Run("returns empty slice when input txs is empty", func(t *testing.T) {
+		txs, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{})
+		require.NoError(t, err)
+		require.Empty(t, txs)
+	})
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, ethTxs, txBytes := generateTxsAndTxsBytes(t, test.depTxNum, test.nonDepTxNum)
+			txs, err := monomer.AdaptPayloadTxsToCosmosTxs(txBytes, test.signTx, test.from)
+			require.NoError(t, err)
+
+			adaptedTxs, err := monomer.AdaptCosmosTxsToEthTxs(txs)
+			require.NoError(t, err)
+			require.Equal(t, len(ethTxs), len(adaptedTxs))
+			for i := range ethTxs {
+				require.Equal(t, ethTxs[i].Hash(), adaptedTxs[i].Hash(), i)
+			}
+		})
+	}
+
+	t.Run("returns error when input contains invalid binary data", func(t *testing.T) {
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{[]byte("invalid")})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unexpected number of messages in Cosmos tx", func(t *testing.T) {
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, &codectypes.Any{}, nil)
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when L1 Attributes tx is not found", func(t *testing.T) {
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{})
+		require.NoError(t, err)
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, msgAny, nil)
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unable to unmarshal MsgL1Txs message", func(t *testing.T) {
```

Comment:
Can you replace all occurrences of `MsgL1Txs` with `MsgApplyL1Txs`?

```suggestion
	t.Run("returns error when unable to unmarshal MsgApplyL1Txs message", func(t *testing.T) {
```

Reply:
Can you please also update the other occurrences of `MsgL1Txs` to be `MsgApplyL1Txs`?

https://github.com/polymerdao/monomer/pull/157/files#diff-94effbf3a78a80713e8f898ba2eb64300db83f4b531c4dcc3d8eed9b1e9c3ca4R181

https://github.com/polymerdao/monomer/pull/157/files#diff-94effbf3a78a80713e8f898ba2eb64300db83f4b531c4dcc3d8eed9b1e9c3ca4R190